tests/tests.rs

```diff
@@ -0,0 +1,336 @@
+
+        let mut cordic = cordic.into_dynamic();
+
+        let sqrt = cordic.run::<Sqrt<N0>>(I1F15::from_num(0.25));
+        debug!("sqrt: {}", sqrt.to_num::<f32>());
+        assert!(is_almost_equals(sqrt.to_num::<f32>(), 0.5));
+        let magnitude = cordic.run::<Magnitude>((I1F15::from_num(0.25), I1F15::from_num(0.5)));
+        debug!("magnitude: {}", magnitude.to_num::<f32>());
+        assert!(is_almost_equals(magnitude.to_num::<f32>(), 0.559));
+    }
+
+    #[test]
+    fn adc() {
+        use super::*;
+
+        // TODO: Is it ok to steal these?
+        let cp = unsafe { stm32::CorePeripherals::steal() };
+        let dp = unsafe { stm32::Peripherals::steal() };
+        let rcc = dp.RCC.constrain();
+        let mut delay = cp.SYST.delay(&rcc.clocks);
+
+        let mut adc = dp
+            .ADC1
+            .claim(adc::ClockSource::SystemClock, &rcc, &mut delay, true);
+
+        adc.enable_temperature(&dp.ADC12_COMMON);
+        adc.enable_vref(&dp.ADC12_COMMON);
+        let sample_time = adc::config::SampleTime::Cycles_640_5;
+
+        let vref = adc.convert(&Vref, sample_time);
+        let vref_cal = VrefCal::get().read();
+        let vdda = VDDA_CALIB * vref_cal as u32 / vref as u32;
+        debug!("vdda: {}mV", vdda);
+        assert!((3200..3400).contains(&vdda));
+
+        let vref = Vref::sample_to_millivolts_ext(vref, vdda, adc::config::Resolution::Twelve);
+        debug!("vref: {}mV", vref);
+        assert!((1182..1232).contains(&vref)); // From G474 datasheet
+
+        let temperature_reading = adc.convert(&Temperature, sample_time);
+        let temp = Temperature::temperature_to_degrees_centigrade(
+            temperature_reading,
+            vdda as f32 / 1000.,
+            adc::config::Resolution::Twelve,
+        );
+        debug!("temp: {}°C", temp);
+        assert!((20.0..30.0).contains(&temp), "20.0 < {} < 30.0", temp);
+    }
+
+    // TODO: This does not seem to work
+    #[test]
+    fn dac() {
+        use super::*;
+
+        // TODO: Is it ok to steal these?
+        let cp = unsafe { stm32::CorePeripherals::steal() };
+        let dp = unsafe { stm32::Peripherals::steal() };
+        let mut rcc = dp.RCC.constrain();
+        let mut delay = cp.SYST.delay(&rcc.clocks);
+
+        let gpioa = dp.GPIOA.split(&mut rcc);
+        let dac1ch1 = dp.DAC1.constrain(gpioa.pa4, &mut rcc);
+
+        let gpioa = unsafe { &*GPIOA::PTR };
+
+        // dac_manual will have its value set manually
+        let mut dac = dac1ch1.calibrate_buffer(&mut delay).enable();
+
+        dac.set_value(0);
+        delay.delay_ms(100);
+        assert!(is_pax_low(&gpioa, 4));
+
+        /*for i in (0..=4095).step_by(10) {
+            dac.set_value(i);
+            delay.delay_ms(1);
+            defmt::println!("i: {}, is_pax_low: {}", i, gpioa.idr().read().bits());
+        }
+
+        delay.delay_ms(100);
+        assert!(!is_pax_low(&gpioa, 4)); // TODO: <---- Why does this not work?
+        */
```

Comment:
Any reason why this does not work?

The pin seems to thinkt that it is low no matter what value I set.

Will try to measure and see what voltage the pin in fact is